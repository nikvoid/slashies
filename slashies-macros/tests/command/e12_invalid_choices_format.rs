use slashies_macros::Command;

/// An invalid command
#[derive(Command)]
#[name = "BadCommand"]
struct BadCommand {
    /// Field with badly formatted choices
    #[choice]
    bad_field: u64,

    /// Ok field
    #[choice(0)]
    #[choice("one", 1)]
    good_field: u64,
}

fn main() {}
