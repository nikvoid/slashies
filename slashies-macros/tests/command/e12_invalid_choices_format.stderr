error: Invalid "choices" attribute. Attribute must be of the form choice(name, value) or choice(value)
 --> tests/command/e12_invalid_choices_format.rs:8:5
  |
8 |     #[choice]
  |     ^^^^^^^^^
